earthfile2llb/interpreter.go

```diff
@@ -376,10 +385,9 @@ func (i *Interpreter) handleForArgs(ctx context.Context, forArgs []string, sl *s
 		return "", nil, i.errorf(sl, "expected IN, got %s", args[1])
 	}
 	variable := args[0]
-	expression := args[2:]
 	runOpts := ConvertRunOpts{
 		CommandName: "FOR",
-		Args:        expression,
+		Args:        []string{strings.Join(args[2:], " ")},
```

Comment:
This feels too arbitrary. The join should only take place in the shell wrappers ideally.

Reply:
I did some more digging and looks like we'll always get `args[2:]` set as [`./cmd`, `"single arg with space"`, ...]. I was worried about the quotes being stripped at this point, but that's not the case, so I'll back out this change.